File: js/sameASCII.js
// Problem# 258
// Return true if the sum of ASCII values of the first string is same as the sum of ASCII values of the second string,
// otherwise return false.
// https://edabit.com/challenge/y6TD7s63sm9RwByLY

const getASCIISum = (str) =>
  [...str].reduce((acc, curr) => acc + curr.charCodeAt(), 0);

const isSameASCII = (a, b) => getASCIISum(a) === getASCIISum(b);

module.exports = isSameASCII;

File: python/same_ascii.py
"""
Same ASCII?

Return True if the sum of ASCII values of the first string is same as the sum of ASCII values of the second string, otherwise return False.

https://edabit.com/challenge/mwNDWczpQ5bm7vLKk
"""


def get_ascii_sum(string: str) -> int:
    return sum(ord(char) for char in string)


def same_ascii(a: str, b: str) -> bool:
    return get_ascii_sum(a) == get_ascii_sum(b)
